### ISSUE TITLE
400, 404 Errors with Python SDK

### ISSUE THREAD

**Issue by lanec** [googleapis/google-api-python-client#468]
I'm experiencing a couple of issues with the API as I move emails between our application and the users inbox.  These issues manifest when I send to a Gmail account from another Gmail account.  However, these issues don't seem to manifest when working with an email generated by Google itself.  I am uploading the sample emails as JSON format to allow others to reproduce the issue.

1. I use the "get" request (https://developers.google.com/gmail/api/v1/reference/users/messages/get) to obtain an email and I store it in a PostgreSQL database in two different columns with both raw and full formats.  
2. I use the "delete" request (https://developers.google.com/gmail/api/v1/reference/users/messages/delete) to remove the email from Gmail. 
3. I use use the raw format of the data along with the "insert" request (https://developers.google.com/gmail/api/v1/reference/users/messages/insert) to place the email back into the users inbox.  

When working with an email sent by Google to a users Gmail account it moves between our software and Gmail without an issue.  

When working with an email I send from my Gmail account to another Gmail account it gives me an error.  If I send the JSON format of the email (the full format) I get a 400 error.  If I send the raw format of the email I get a 404 not found error.  

Exact Errors: 
404 - <HttpError 404 when requesting https://www.googleapis.com/gmail/v1/users/me/messages?alt=json returned "Not Found">
400 - <HttpError 400 when requesting https://www.googleapis.com/gmail/v1/users/me/messages?alt=json returned "'raw' RFC822 payload message string or uploading message via /upload/* URL required">

I'm using Python3.6 and Google API Client 1.6.4

[emails.zip](https://github.com/google/google-api-python-client/files/1640586/emails.zip)

**Comment by lanec** [comment]
Hi @mcdonc 

I shared the code snippets from the Google API documentation.  

The application I am working on is proprietary but the code snippets that Google shares is verbatim what I was using.

**Comment by mcdonc** [comment]
@lanec ok, I'll eventually try to divine what you're doing from your English description.  This issue will be closed faster, however, if someone is able to supply a script that demonstrates the issue so diviniation is unneccessary. :)

**Comment by lanec** [comment]
@mcdonc I don't blame you for not being able to make progress on this in a timely manner.  

It was frustrating writing software to interface with this API since the Google documentation is very much out of date. Even determining the dependencies for authenticating with them was a project all in of itself.   

The project is proprietary so I can't share it on here.  

I provided links to the specific requests being made and the sample emails.  If you authenticate with Google and make the requests using the sample emails then you will get the errors I got.  

Unfortunately, someone at Google really needs to update the Google API python documentation.
